filebeat/beater/filebeat.go

```diff
@@ -195,14 +196,16 @@ func (fb *Filebeat) setupPipelineLoaderCallback(b *beat.Beat) error {
 
 	overwritePipelines := true
 	b.OverwritePipelinesCallback = func(esConfig *conf.C) error {
-		esClient, err := eslegclient.NewConnectedClient(esConfig, "Filebeat")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		esClient, err := eslegclient.NewConnectedClient(ctx, esConfig, "Filebeat")
 		if err != nil {
 			return err
 		}
 
 		// When running the subcommand setup, configuration from modules.d directories
 		// have to be loaded using cfg.Reloader. Otherwise those configurations are skipped.
-		pipelineLoaderFactory := newPipelineLoaderFactory(b.Config.Output.Config())
+		pipelineLoaderFactory := newPipelineLoaderFactory(ctx, b.Config.Output.Config())
```

Comment:
Why context is needed here? (Sorry as not so familiar with this part of code) I see that pipelineLoaderFactory is called inside NewFactory. So is not better to pass the context there?

Reply:
This context controls the lifecycle of the connection used to create the pipeline. It ensures that by the time the callback is done, all connections/requests it created are also shutdown.

The context is passed to `newPipelineLoaderFactory`
https://github.com/belimawr/beats/blob/cfd1f1cd26450e2c77d3f73f0656caa76f183ff1/filebeat/beater/filebeat.go#L205 and then passed to `eslegclient.NewConnectedClient` https://github.com/belimawr/beats/blob/cfd1f1cd26450e2c77d3f73f0656caa76f183ff1/filebeat/beater/filebeat.go#L492